lib/widgets/compose_box.dart

```diff
@@ -1023,6 +1023,19 @@ class ComposeBox extends StatelessWidget {
   final GlobalKey<ComposeBoxController>? controllerKey;
   final Narrow narrow;
 
+  static bool hasComposeBox(Narrow narrow) {
+    switch (narrow) {
+      case CombinedFeedNarrow():
+      case MentionsNarrow():
+        return false;
+
+      case StreamNarrow():
```

Comment:
nit: order this switch the same way as the one in `build` just below — this is effectively a summary of that one and needs to be kept in sync, so keeping the same ordering helps with verifying that at a glance